src/IssueViz/Editor/SelectedIssueTagging/Buffer/SelectedIssueLocationTagger.cs

```diff
@@ -0,0 +1,86 @@
+ * version 3 of the License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with this program; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+ */
+
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using Microsoft.VisualStudio.Text;
+using Microsoft.VisualStudio.Text.Tagging;
+using SonarLint.VisualStudio.IssueVisualization.Editor.LocationTagging;
+using SonarLint.VisualStudio.IssueVisualization.Models;
+using SonarLint.VisualStudio.IssueVisualization.Selection;
+
+namespace SonarLint.VisualStudio.IssueVisualization.Editor.SelectedIssueTagging.Buffer
+{
+    internal class SelectedIssueLocationTagger: FilteringTaggerBase<IIssueLocationTag, ISelectedIssueLocationTag>
+    {
+        private readonly IAnalysisIssueSelectionService issueSelectionService;
+
+        private bool disposedValue;
+
+        public SelectedIssueLocationTagger(ITagAggregator<IIssueLocationTag> tagAggregator, ITextBuffer textBuffer, IAnalysisIssueSelectionService issueSelectionService)
+            : base(tagAggregator, textBuffer)
+        {
+            this.issueSelectionService = issueSelectionService;
+
+            // Changing any of the selected issue/flow/location will always result in the
+            // "SelectedLocationChanged" event being raised
+            issueSelectionService.SelectionChanged += OnSelectionChanged;
+        }
+
+        private void OnSelectionChanged(object sender, SelectionChangedEventArgs e)
+        {
+            if (e.SelectionChangeLevel == SelectionChangeLevel.Flow || e.SelectionChangeLevel == SelectionChangeLevel.Issue)
+            {
+                UpdateWholeSpan();
+            }
+        }
+
+        #region ITagger methods
+
+        protected override TagSpan<ISelectedIssueLocationTag> CreateTagSpan(IIssueLocationTag trackedTag, NormalizedSnapshotSpanCollection spans)
+        {
+            return new TagSpan<ISelectedIssueLocationTag>(trackedTag.Location.Span.Value, new SelectedIssueLocationTag(trackedTag.Location));
+        }
+
+        protected override IEnumerable<IMappingTagSpan<IIssueLocationTag>> Filter(IEnumerable<IMappingTagSpan<IIssueLocationTag>> trackedTagSpans)
+        {
+            var selectedSecondaryLocVizs = issueSelectionService.SelectedFlow?.Locations;
+            if (selectedSecondaryLocVizs == null || selectedSecondaryLocVizs.Count == 0) { return Array.Empty<IMappingTagSpan<IIssueLocationTag>>(); }
+
+            // Filter to any selected locations.
+            // There could be multiple secondary issues locations in this file for different issues.
```

Comment:
Misc: the last comment sounds misleading, since it sounds like "filter the locations and _show_ secondary locations for different issues" -- might be worth just switching them, e.g.
"There could be multiple secondary issues locations in this file for different issues so we display only the secondary locations of currently selected flow".
(or maybe remove the comment, since that's what line 65 means)

Reply:
I'll remove the comment.